internal/executor/instance/persistentworker_test.go

```diff
@@ -0,0 +1,28 @@
+package instance_test
+
+import (
+	"context"
+	"github.com/cirruslabs/cirrus-cli/internal/executor/instance"
+	"github.com/stretchr/testify/assert"
+	"runtime"
+	"testing"
+	"time"
+)
+
+func TestRetrieveAgentBinary(t *testing.T) {
+	// Does it work?
+	_, err := instance.RetrieveAgentBinary(context.Background(), instance.AgentVersion, runtime.GOOS, runtime.GOARCH)
+	if err != nil {
+		t.Fatal()
+	}
+
+	// Does it cache the agent?
+	cachedRetrievalStart := time.Now()
+
+	_, err = instance.RetrieveAgentBinary(context.Background(), instance.AgentVersion, runtime.GOOS, runtime.GOARCH)
+	if err != nil {
+		t.Fatal()
+	}
+
+	assert.WithinDuration(t, cachedRetrievalStart, time.Now(), 100*time.Millisecond)
```

Comment:
Maybe check that file creation timestamp is the same as before?

Reply:
Added more checks in https://github.com/cirruslabs/cirrus-cli/pull/139/commits/c1fd9ee0f339a499457e5a5fa8a07bffba71cb0a.